src/steps/object-field-equals.ts

```diff
@@ -0,0 +1,108 @@
+    description: 'Salesforce object name',
+  }, {
+    field: 'id',
+    type: FieldDefinition.Type.STRING,
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+    description: 'Object ID',
+  }, {
+    field: 'operator',
+    type: FieldDefinition.Type.STRING,
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+    description: 'Check Logic (be, not be, contain, not contain, be greater than, or be less than)',
+  }, {
+    field: 'expectedValue',
+    type: FieldDefinition.Type.ANYSCALAR,
+    description: 'Expected field value',
+  }];
+  protected expectedRecords: ExpectedRecord[] = [{
+    id: 'salesforceObject',
+    type: RecordDefinition.Type.KEYVALUE,
+    fields: [{
+      field: 'Id',
+      type: FieldDefinition.Type.NUMERIC,
+      description: "Object's SalesForce ID",
+    }, {
+      field: 'CreatedDate',
+      type: FieldDefinition.Type.DATETIME,
+      description: "Object's Created Date",
+    }, {
+      field: 'LastModifiedDate',
+      type: FieldDefinition.Type.DATETIME,
+      description: "Object's Last Modified Date",
+    }],
+    dynamicFields: true,
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const field: string = stepData.field;
+    const objName: string = stepData.objName;
+    const id: string = stepData.id;
+    const operator: string = stepData.operator || 'be';
+    const expectedValue: string = stepData.expectedValue;
+    let object: Record<string, any>;
+
+    try {
+      object = await this.client.findObjectById(objName, id);
+    } catch (e) {
+      return this.error('There was a problem checking the Object: %s', [e.toString()]);
+    }
+
+    try {
+      if (!object) {
+        // If no results were found, return an error.
+        return this.error('No Object found with id %s', [id]);
+      }
+
+      const record = this.createRecord(object);
+
+      if (!object.hasOwnProperty(field)) {
+        // If the given field does not exist on the user, return an error.
+        return this.error('The %s field does not exist on Object %s', [field, id], [record]);
+      } else if (this.compare(operator, object[field], expectedValue)) {
+        // If the value of the field matches expectations, pass.
+        return this.pass(this.operatorSuccessMessages[operator], [field, expectedValue], [record]);
+      } else {
+        // If the value of the field does not match expectations, fail.
+        return this.fail(this.operatorFailMessages[operator], [field, expectedValue, object[field]], [record]);
+      }
+    } catch (e) {
+      if (e instanceof util.UnknownOperatorError) {
+        return this.error('%s Please provide one of: %s', [e.message, baseOperators.join(', ')]);
+      }
+      if (e instanceof util.InvalidOperandError) {
+        return this.error(e.message);
+      }
+      return this.error('There was an error during validation of object field: %s', [e.message]);
```

Comment:
- [ ] Same as above: consider adding the `objName` to make the message friendlier.